core/src/test/scala/cats/derived/adtdefns.scala

```diff
@@ -55,8 +55,14 @@ object TestDefns {
   final case class Recursive(i: Int, is: Option[Recursive])
   object Recursive extends ((Int, Option[Recursive]) => Recursive) {
 
-    implicit val arbitrary: Arbitrary[Recursive] =
-      Arbitrary(Gen.delay(Arbitrary.arbitrary[(Int, Option[Recursive])].map(tupled)))
+    implicit val arbitrary: Arbitrary[Recursive] = {
+      def recursive(maxLength: Int): Gen[Recursive] = for {
+        i <- Arbitrary.arbitrary[Int]
+        n <- Gen.choose(0, maxLength - 1)
+        is <- if (n <= 0) Gen.const(None) else Gen.option(recursive(n))
+      } yield Recursive(i, is)
+      Arbitrary(recursive(32))
```

Comment:
Oh, I see.  You've dropped `Gen.sized` and just hard-coded the depth.

Reply:
Yes but I like your suggestion more.